Nos/Service/DependencyInjection.swift

```diff
@@ -84,6 +84,11 @@ extension DependencyValues {
         get { self[KeychainKey.self] }
         set { self[KeychainKey.self] = newValue }
     }
+
+    var refreshController: RefreshController {
```

Comment:
I think we should reserve dependency injection for global side-effecty things. I think we are going to want to instantiate more RefreshControllers i.e. if we add this "new notes" notification to the profile screen. Can we instantiate it as `@State` in the view like we do the `ContentWarningController`?

Reply:
🤔 I see your next ticket is adding this to the profile screen. Maybe it makes more sense to do this in that PR?

Reply:
Nah, I'll take care of it now.

Reply:
I think the reason I put it here is that I see all external objects as dependencies. I think I learned this from Martin Fowler's [Dependency Injection article](https://martinfowler.com/articles/injection.html) years and years and years ago, and maybe even misunderstood exactly what he meant. But I digress. It's clear that our friends at Point-Free have a different [definition](https://pointfreeco.github.io/swift-dependencies/main/documentation/dependencies/whataredependencies):

> Dependencies in an application are the types and functions that need to interact with outside systems that you do not control.

So I assume we're all in agreement that a type shouldn't instantiate its own dependencies, which explains why I put the `RefreshController` here. I was thinking this is the place to put dependencies but had a very different definition of dependencies.

I do agree that we're going to want more than one instance of a `RefreshController`. I'm not totally understanding how our Dependencies framework works, though. I had the `@Dependency` in both `HomeFeedView` and `ProfileView` and saw everything working as expected -- when I refreshed the `HomeFeedView` with its `refreshController`, the `ProfileView` did not get refreshed. The same was true in reverse -- refreshing the `ProfileView` did not cause the `HomeFeedView` to refresh. I did a little [reading](https://pointfreeco.github.io/swift-dependencies/main/documentation/dependencies/lifetimes) this morning and understand that `@Dependency` works like `@TaskLocal`, so I assume I'm just not fully grasping how that works.

Whoops, this turned into an essay. I think the main thing I'm wondering is: how would we test `HomeFeedView` and its interaction with `RefreshController` if it's a `@State` instead of `@Dependency`? Maybe we wouldn't test it anyway -- as you can see, I didn't -- but I like to at least think about it and try to make things testable whenever possible.

Reply:
🤔 maybe I don't understand how the Dependencies library works. I thought if `HomeFeedView` and `ProfileView` shared the same `@Dependency var refreshController` than refreshing one would refresh the other. That's how it works for `CurrentUser` etc.

I think if we go to the point where we wanted to write an integration test for how `HomeFeedView` and `RefreshController` work together then we would change `@State var refreshController` to `@Binding var refreshController` and allow the parent view to pass a controller in. 

Although in general my strategy for view testing has been to use SwiftUI Previews as proof that Views are properly wired up to their sources of truth.